testsuite/MDAnalysisTests/coordinates/test_rdkit.py

```diff
@@ -20,59 +20,480 @@
+        expected = np.empty((1, u.atoms.n_atoms, 3), dtype=np.float32)
         expected[:] = np.nan
         assert_equal(u.trajectory.coordinate_array, expected)
 
     def test_compare_mol2reader(self):
-        universe = mda.Universe(mol2_mol())
+        universe = mda.Universe(MolFactory.mol2_mol())
         mol2 = mda.Universe(mol2_molecule)
         assert universe.trajectory.n_frames == mol2.trajectory.n_frames
-        assert_equal(universe.trajectory.ts.positions, 
+        assert_equal(universe.trajectory.ts.positions,
                      mol2.trajectory.ts.positions)
-        
+
+
+@requires_rdkit
+class TestRDKitConverter(object):
+    @pytest.fixture
+    def pdb(self):
+        return mda.Universe(PDB_full)
+
+    @pytest.fixture
+    def mol2(self):
+        u = mda.Universe(mol2_molecule)
+        # add elements
+        elements = np.array([guess_atom_element(x) for x in u.atoms.types],
+                            dtype=object)
+        u.add_TopologyAttr('elements', elements)
+        return u
+
+    @pytest.fixture
+    def peptide(self):
+        u = mda.Universe(GRO)
+        elements = mda.topology.guessers.guess_types(u.atoms.names)
+        u.add_TopologyAttr('elements', elements)
+        return u.select_atoms("resid 2-12")
+
+    @pytest.mark.parametrize("smi", ["[H]", "C", "O", "[He]"])
+    def test_single_atom_mol(self, smi):
+        u = mda.Universe.from_smiles(smi, addHs=False,
+                                     generate_coordinates=False)
+        mol = u.atoms.convert_to("RDKIT")
+        assert mol.GetNumAtoms() == 1
+
+    @pytest.mark.parametrize("resname, n_atoms, n_fragments", [
+        ("PRO", 14, 1),
+        ("ILE", 38, 1),
+        ("ALA", 20, 2),
+        ("GLY", 21, 3),
+    ])
+    def test_mol_from_selection(self, peptide, resname, n_atoms, n_fragments):
+        mol = peptide.select_atoms("resname %s" % resname).convert_to("RDKIT")
+        assert n_atoms == mol.GetNumAtoms()
+        assert n_fragments == len(Chem.GetMolFrags(mol))
+
+    @pytest.mark.parametrize("sel_str, atom_index", [
+        ("resid 1", 0),
+        ("resname LYS and name NZ", 1),
+        ("resid 34 and altloc B", 2),
+    ])
+    def test_monomer_info(self, pdb, sel_str, atom_index):
+        sel = pdb.select_atoms(sel_str)
+        umol = sel.convert_to("RDKIT")
+        atom = umol.GetAtomWithIdx(atom_index)
+        mda_index = atom.GetIntProp("_MDAnalysis_index")
+        mi = atom.GetMonomerInfo()
+
+        for mda_attr, rd_attr in RDATTRIBUTES.items():
+            rd_value = getattr(mi, "Get%s" % rd_attr)()
+            if hasattr(sel, mda_attr):
+                mda_value = getattr(sel, mda_attr)[mda_index]
+                if mda_attr == "names":
+                    rd_value = rd_value.strip()
+                assert rd_value == mda_value
+
+    @pytest.mark.parametrize("rdmol", ["mol2_mol", "smiles_mol"],
+                             indirect=True)
+    def test_identical_topology(self, rdmol):
```

Comment:
Would it be worth testing the full roundtrip too? i.e. MDA->RDKit->MDA

Reply:
Good idea -- a general acid test for converters!

Reply:
You mean something like this ?
```python
u = mda.Universe(...)
mol = u.atoms.convert_to("RDKIT")
umol = mda.Universe(mol)
assert u == umol
```
I'm not sure how Universe comparison works

Reply:
Yes, that would be the idea way – but I also don't think that we have `Universe.__eq__` defined...  so comparing topology things (list of atoms, bonds) (`assert_equal` for exact array comparisons) and positions (`assert_almost_equal` for floating point numbers) should do the job.

(Or raise an issue to make `Universe.__eq__` work, which will likely lead to having to make `Topology.__eq__` and `Reader.__eq__` work... at least `Timestep.__eq__` exists already.)